src/main/java/org/mybatis/caches/redis/KryoSerializer.java

```diff
@@ -31,9 +31,14 @@
   // Enum singleton, which is preferred approach since Java 1.5
   INSTANCE;
 
-  private Kryo kryo;
-  private Output output;
-  private Input input;
+  /**
+   * kryo is thread-unsafe, use ThreadLocal.
+   */
+  private ThreadLocal<Kryo> kryos = ThreadLocal.withInitial(() -> {
+    Kryo kryo = new Kryo();
```

Comment:
@a1364533187 Can you please just return new Kryo() here instead of creating new object first?

Reply:
I can merge it in then.  Thanks.

Reply:
just return new Kryo() here, I fix it, thank you